### DISCUSSION TITLE
Do you need tidy3d components to be hashable for gdsfactory?

### DISCUSSION THREAD

**Discussion by tylerflex** [gdsfactory/gdsfactory#1542] [General]
Just a question, do you have opinions on whether tidy3d components, eg Simulation, should be hashable? we started introducing more native np.ndarray support internally, which makes the objects no longer hashable unless we implement custom hash functions. Just wanted to check if you had a preference on this out of curiosity.

**Comment by joamatab** [comment]
We hash the serialization of the simulation (json) for identification of the simulation 

As long as that still works is fine with us

**Reply by tylerflex** [reply]
Yea that should always work almost by definition. If that's the case, I think maybe we can just use `np.ndarray` internally for ArrayLike ( @lucas-flexcompute ). By the way, how do you generate this json? through `obj.json()` using pydantic? Note that we have a (private) method `obj._json_string()` that generates what we actually submit, which takes care of Infinity values.
